code/game/objects/items/weapons/RSF.dm

```diff
@@ -67,18 +67,24 @@ RSF
 		to_chat(user, "The RSF can only create service items on tables, or floors.")
 		return
 
-	to_chat(user, "Dispensing " + configured_items[mode][1] + "...")
-	playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
-	var type_path = configured_items[mode][3]
-	new type_path(spawn_location)
 	if(isrobot(user))
 		var/mob/living/silicon/robot/engy = user
+		if (engy.cell.charge < configured_items[mode][2])
+			to_chat(user, "<span class='warning'>Insufficient energy.</span>")
+			return
 		engy.cell.charge -= configured_items[mode][2]
 	else
+		if (matter == 0)
+			to_chat(user, "<span class='warning'>Insufficient matter.</span>")
+			return
 		matter--
 		to_chat(user, "The RSF now holds [matter]/30 fabrication-units.")
 		desc = "A RSF. It currently holds [matter]/30 fabrication-units."
 
+	to_chat(user, "Dispensing " + configured_items[mode][1] + "...")
+	playsound(src.loc, 'sound/machines/click.ogg', 10, 1)
+	var type_path = configured_items[mode][3]

```

Comment:
what are you doinggggg

Reply:
Hey, this one isn't mine. I just moved it. It was already that way!
But yeah, changed.